### PR TITLE
Feature/runtime tests

### PR DESCRIPTION
**Pull request by KevReed**

### CODE REVIEW

File: src/DSharp.Script/test/array.spec.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+describe('array', () => {
+    test('null returns null', () => {
+const array = require('../dist/ss.js').array;
```
Comment (fred-perkins) [inline src/DSharp.Script/test/array.spec.js]: Formatting seems off on the const here

File: src/DSharp.Script/test/array.spec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        expect(array(null)).toBeNull();
+    });
+
+    test('arguments is converted to array', () => {
+const array = require('../dist/ss.js').array;
```
Comment (fred-perkins) [inline src/DSharp.Script/test/array.spec.js]: and here

File: src/DSharp.Script/test/array.spec.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        expect(Array.isArray(arguments)).toBe(false);
+        expect(Array.isArray(array(arguments))).toBe(true);
+    });
+
+    test('array is cloned', () => {
+const array = require('../dist/ss.js').array;
```
Comment (fred-perkins) [inline src/DSharp.Script/test/array.spec.js]: Here too

File: src/DSharp.Script/test/clearKeys.spec.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+
+describe('clearKeys', () => {
+    test('missing key returns false', () => {
+        const clearKeys = require('../dist/ss.js').clearKeys;
```
Comment (fred-perkins) [inline src/DSharp.Script/test/clearKeys.spec.js]: Is there a way we can move `'../dist/ss.js'` to a const somewhere for all specs to use?

File: src/DSharp.Script/DSharp.Scripts.csproj [hunk 5]
```diff
@@ -1,25 +1,67 @@
...
     <DisableFastUpToDateCheck>true</DisableFastUpToDateCheck>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <JavaScriptTestRoot>.\test</JavaScriptTestRoot>
+    <JavaScriptTestFramework>Jest</JavaScriptTestFramework>
+    <GenerateProgramFile>false</GenerateProgramFile>
   </PropertyGroup>
-  
+
+  <ItemGroup>
+    <!--<ProjectCapability Include="DotNetCoreWeb" />-->
```
Comment (fred-perkins) [inline src/DSharp.Script/DSharp.Scripts.csproj]: Remove the comment

File: src/DSharp.Script/DSharp.Scripts.csproj [hunk 6]
```diff
@@ -1,25 +1,67 @@
...
+  <ItemGroup>
+    <SourceScript Include="src\**\*.js" />
+    <Content Include="src\**\*.js">
+      <TestFramework>Jest</TestFramework>
+    </Content>
+  </ItemGroup>
+
   <ItemGroup>
-    <PackageReference Include="Microsoft.TypeScript.MSBuild" Version="2.9.1" />
+    <None Remove="test\array.spec.js" />
```
Comment (fred-perkins) [inline src/DSharp.Script/DSharp.Scripts.csproj]: Worth doing this as a recursive property:
`<None Remove="test\*.spec.js" />`

File: src/DSharp.Script/DSharp.Scripts.csproj [hunk 7]
```diff
@@ -1,25 +1,67 @@
...
+    <None Remove="test\keyExists.spec.js" />
+    <None Remove="test\keys.spec.js" />
+    <None Remove="test\value.spec.js" />
   </ItemGroup>
 
-  <Target Name="RunScriptBuild" AfterTargets="Build">
-    <RemoveDir Directories="dist" />
-    <Message Importance="high" Text="Restoring dependencies using 'npm'. This may take several minutes..." />
+  <ItemGroup>
+    <Content Include="test\array.spec.js" />
```
Comment (fred-perkins) [inline src/DSharp.Script/DSharp.Scripts.csproj]: Worth doing this as a recursive property:
`<Content Include="test\*.spec.js" />`

File: src/DSharp.Script/DSharp.Scripts.nuspec [hunk 8]
```diff
@@ -5,7 +5,7 @@
     <authors>$authors$</authors>
     <version>$version$</version>
     <copyright>$copyright$</copyright>
-    <description>Provides the DSharp runtime library</description>
+    <description>the DSharp runtime library</description>
```
Comment (fred-perkins) [inline src/DSharp.Script/DSharp.Scripts.nuspec]: Capitial `T` please!

File: src/DSharp.Script/package.json [hunk 9]
```diff
@@ -1,16 +1,20 @@
 {
   "name": "dsharp.script",
   "version": "1.0.0",
-  "description": "",
+  "description": "the DSharp runtime library",
```
Comment (fred-perkins) [inline src/DSharp.Script/package.json]: Capital `T` please

File: src/DSharp.Script/DSharp.Scripts.csproj (hunk 7 above)
Comment (KevReed) [reply src/DSharp.Script/DSharp.Scripts.csproj]: visual studio likes to expand wildcards when used on the standard itemgroups

Comment (fred-perkins) [reply src/DSharp.Script/DSharp.Scripts.csproj]: Not in the new SDK :)

File: src/DSharp.Script/test/enumerate.spec.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
+
+describe('enumerate', () => {
+    test('null returns empty enumerator', () => {
+        //Arrange
+        const enumerate = require(process.env['RUNTIME']).enumerate;
+
+        //Act
+        const result = enumerate(null);
+        expect(result.moveNext()).toBe(false);
```
Comment (isc30) [inline src/DSharp.Script/test/enumerate.spec.js]: assert :P

**Pull request merged**